src/serialization/deserializer.rs

```diff
@@ -30,10 +30,44 @@ pub enum DeserializeError {
 
 impl std::convert::From<std::io::Error> for DeserializeError {
     fn from(e: std::io::Error) -> Self {
-        DeserializeError::IoError(e.kind())
+        DeserializeError::IoError(e)
     }
 }
 
+impl fmt::Display for DeserializeError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            DeserializeError::IoError(e) => write!(f, "An i/o operation failed: {}", e),
+            DeserializeError::InvalidCookie => write!(
+                f,
+                "The cookie (first 4 bytes) did not match that for any supported format"
+            ),
+            DeserializeError::UnsupportedFeature => write!(
+                f,
+                "The histogram uses features that this implementation doesn't support"
+            ),
+            DeserializeError::UnsuitableCounterType => write!(
+                f,
+                "A count exceeded what can be represented in the chosen counter type"
+            ),
+            DeserializeError::InvalidParameters => write!(
+                f,
+                "The serialized parameters were invalid(e.g. lowest value, highest value, etc)"
+            ),
+            DeserializeError::UsizeTypeTooSmall => write!(
+                f,
+                "The current system's pointer width cannot represent the encoded histogram"
+            ),
+            DeserializeError::EncodedArrayTooLong => write!(
+                f,
+                "The encoded array is longer than it should be for the histogram's value range"
+            ),
+        }
+    }
+}
+
+impl error::Error for DeserializeError {}
```

Comment:
This should override the implementation of [`Error::source`](https://doc.rust-lang.org/std/error/trait.Error.html#method.source).